domene/src/main/kotlin/no/nav/tiltakspenger/saksbehandling/domene/vilkår/livsopphold/LivsoppholdVilkaar.kt

```diff
@@ -0,0 +1,118 @@
+                        command.ytelseForPeriode.map { PeriodeMedVerdi(it.tilYtelse(), it.periode) },
+                    ).utvid(HarYtelse.HAR_IKKE_YTELSE, vurderingsperiode),
+                    årsakTilEndring = command.årsakTilEndring,
+                    saksbehandler = command.saksbehandler,
+                    tidsstempel = LocalDateTime.now(),
+                )
+                return this.copy(
+                    aapDelVilkår = aapDelVilkår.leggTilSaksopplysning(saksopplysning),
+                    utfall = vurderMaskinelt(aapDelVilkår, alderspensjonDelVilkår),
+                )
+            }
+
+            Livsoppholdsytelse.DAGPENGER -> TODO()
+            Livsoppholdsytelse.ALDERSPENSJON -> TODO()
+            Livsoppholdsytelse.GJENLEVENDEPENSJON -> TODO()
+            Livsoppholdsytelse.SYKEPENGER -> TODO()
+            Livsoppholdsytelse.JOBBSJANSEN -> TODO()
+            Livsoppholdsytelse.FORELDREPENGER -> TODO()
+            Livsoppholdsytelse.OMSORGSPENGER -> TODO()
+            Livsoppholdsytelse.OPPLÆRINGSPENGER -> TODO()
+        }
+    }
+
+    init {
+        require(vurderingsperiode == aapDelVilkår.vurderingsperiode) {
+            "vurderingsperioden ($vurderingsperiode) og delvilkåret for aap sin vurderingsperiode(${aapDelVilkår.vurderingsperiode}) må være like."
+        }
+        require(vurderingsperiode == alderspensjonDelVilkår.vurderingsperiode) {
+            "vurderingsperioden ($vurderingsperiode) og delvilkåret for alderspensjon sin vurderingsperiode(${aapDelVilkår.vurderingsperiode}) må være like."
+        }
+        require(vurderingsperiode == utfall.totalePeriode) {
+            "vurderingsperioden ($vurderingsperiode) og utfallet sin totale periode(${utfall.totalePeriode}) må være like."
+        }
+    }
+
+    companion object {
+
+        fun opprett(
+            vurderingsperiode: Periode,
+            alderspensjonSøknadSaksopplysning: AlderspensjonSaksopplysning.Søknad,
+            // TODO: Her kommer alle saksopplysningene fra søknaden
+        ): LivsoppholdVilkaar {
+            val aapDelVilkår = AAPDelVilkaar(vurderingsperiode)
+            val alderspensjonDelVilkår = AlderspensjonDelVilkaar(vurderingsperiode, alderspensjonSøknadSaksopplysning)
+            return LivsoppholdVilkaar(
+                vurderingsperiode = vurderingsperiode,
+                aapDelVilkår = AAPDelVilkaar(vurderingsperiode),
+                alderspensjonDelVilkår = AlderspensjonDelVilkaar(vurderingsperiode, alderspensjonSøknadSaksopplysning),
+                utfall = vurderMaskinelt(aapDelVilkår, alderspensjonDelVilkår),
+            )
+        }
+
+        fun vurderMaskinelt(
+            aapDelVilkår: AAPDelVilkaar,
+            alderspensjonDelVilkår: AlderspensjonDelVilkaar,
+        ): Periodisering<Utfall2> =
+            listOf(aapDelVilkår.utfall, alderspensjonDelVilkår.utfall).reduser(::kombinerDelVilkår)
+
+        fun kombinerDelVilkår(utfall1: Utfall2, utfall2: Utfall2): Utfall2 =
```

Comment:
På dette nivået vil vi vel bruke SamletUtfall? Er vel kun på den innerste periodiseringen vi vil bruke Utfall2?

Reply:
Vi vil vel ha begge deler? Det er ikke tilstrekkelig med et samletUtfall, vi må jo kunne kombinere de ulike inngangsvilkårene for å komme frem til et samlet utfall på tvers av inngangsvilkårene.